### ISSUE TITLE
Will this work with covr() ?

### ISSUE THREAD

**Issue by MarkEdmondson1234** [rstudio/shinytest#97]
Hi, great idea I've tried to get some testing together with RSelenium before but it never quite worked, looking forward to using this. 

I just wondered if you knew if this package will work with https://github.com/jimhester/covr ?  I'd like to be able to see which code ran in tests, say for RStudio Addins and some specific Shiny modules within packages. At the moment I have to ignore them in testthat, for example here: https://codecov.io/gh/MarkEdmondson1234/googleAuthR/src/7605b8c5b6f5956683735020a9c080f901fdada6/R/googleAuthR_shiny.R

**Comment by jimhester** [comment]
I spent some time getting covr to work with shiny apps in general and in shinytest in particular. The bottom line is I have a proof of concept, however it would need more work to make it a seamless experience for the user, and input on the user API would be appreciated.

Using [jimhester/covr@shiny-app](https://github.com/jimhester/covr/compare/shiny-app?expand=1) and [jimhester/shinytest@covr](https://github.com/rstudio/shinytest/compare/master...jimhester:covr?expand=1). The changes in shinytest are basically to add the covr instrumentation to the application before running it, and register a finalizer which can be called to dump the counts after the process exists. It also needs to explicitly signal the process to end with `SIGINT` rather than killing it with `SIGKILL`, so the R process can exit cleanly (and call the finalizer). I also needed to explicitly use `app$stop()` in the test script to get the signal to run, it was not clear to me where the shiny process is stopped when this is not done.

The bottom line is it is technically possible without major changes in either covr or shinytest, but needs further work for a clean interface and user experience.

**Comment by MarkEdmondson1234** [comment]
Wonderful news!  Happy to test anything out in the wild if needed, just point me to an example.

**Comment by bedantaguru** [comment]
Any progress on this. Seems like a great feature to have. @jimhester shall I test it out as suggested by you? 

I think this is related to https://github.com/r-lib/covr/issues/277

**Comment by hadley** [comment]
I think this is mostly remedied by the introduction of `testServer()` in shiny 1.5.0 — that way you can test your reactivity using regular tests (https://mastering-shiny.org/scaling-testing.html#testing-reactivity) with which covr already works. Then you only need shinytest for tests of javascript, which covr isn't going to help with anyway (at least not without a very large amount of work).
